### ISSUE TITLE
Documentation on the cases where funTreeless is Nothing.

### ISSUE THREAD

**Issue by xekoukou** [agda/agda#3263]
While trying to compile the ```Prelude``` Library, I stumbled upon a [function](https://github.com/UlfNorell/agda-prelude/blob/master/src/Prelude/Equality.agda#L87) that did not return a treeless Term.

Is it safe to ignore it?
The haskell backend apparently ignores [them](https://github.com/agda/agda/blob/323f58f9b8dad239142ed1dfa0c60338ea2cb157/src/full/Agda/Compiler/MAlonzo/Compiler.hs#L366)
edit : (it seems that toTreeless computes the term even if it does not exist.)

There are more missing values like funMutual which I need (because OCaml requires to explicitly tell
that your function is recursive.)

Maybe we could add documentation about it.
```
Defn {
  defArgInfo        =
    ArgInfo {argInfoHiding = NotHidden, argInfoModality = Modality {modRelevance = Relevant, modQuantity = Quantityω}, argInfoOrigin = UserWritten, argInfoFreeVariables = UnknownFVs}
  defName           = Prelude.Equality._∎
  defType           =
    {a : Agda.Primitive.Level} ->
      {A : Set @0} ->
        (x : @0) -> Agda.Builtin.Equality._≡_ {@2} {@1} @0 @0
  defPolarity       = [Invariant,Invariant,Invariant]
  defArgOccurrences = [Unused,Unused,Unused]
  defDisplay        = []
  defMutual         = MutId 18
  defCompiledRep    = fromList []
  defInstance       = Nothing
  defCopy           = False
  defMatchable      = False
  defInjective      = False
  theDef            =
    Function {
      funClauses      =
        {a : Agda.Primitive.Level} {A : Set @0} (x : @0) |-
          {a = _@2} {A = _@1} (x = x@0) =
            Agda.Builtin.Equality._≡_.refl :
              Agda.Builtin.Equality._≡_ {@2} {@1} @0 @0
      funCompiled     =
        Just (Done [ωri{"_"},ωri{"_"},ωru("x")] (Con (ConHead {conName = Agda.Builtin.Equality._≡_.refl, conInductive = inductive, conFields = []}) ConOCon []))
      funTreeless     = Nothing
      funInv          = NotInjective
      funMutual       = Nothing
      funAbstr        = ConcreteDef
      funDelayed      = NotDelayed
      funProjection   = Nothing
      funFlags        = fromList [FunInline]
      funTerminates   = Just True
      funWith         = Nothing
      funCopatternLHS = False
      } }
```

**Comment by xekoukou** [comment]
[ defNoCompilation](https://github.com/agda/agda/blob/master/src/full/Agda/TypeChecking/Monad/Base.hs#L1425) should be set to true if a specific function is not to be compiled.

Simpler example : 
```agda

data Bat : Set where
  bat : Bat

fun : Bat -> Bat
fun x = bat
```

**Comment by xekoukou** [comment]
It seems from the code that ```funMutual``` is only updated on specific cases only, when we have mutual 
blocks. In a simple recursive function it is not. I am closing the issue.

**Comment by gallais** [comment]
Might still be worth adding documentation (where `Defn` is defined).

**Comment by andreasabel** [comment]
The commit mentioned above actually fixes #3262, not this one.
